internal/cephfs/clone.go

```diff
@@ -95,10 +96,14 @@ func createCloneFromSubvolume(ctx context.Context, volID, cloneID volumeID, volO
 
 	cloneState, cloneErr := volOpt.getCloneState(ctx, cloneID)
 	if cloneErr != nil {
+		util.ErrorLog(ctx, "failed to get clone state: %v", cloneErr)
 		return cloneErr
 	}
 
 	switch cloneState {
+	case cephFSCloneError:
+		util.ErrorLog(ctx, "clone is invalid for %v", cloneID)
+		return ErrInvalidClone
```

Comment:
These cases and the ones in `createCloneFromSnapshot()` seem very similar.

You can reduce the code duplication (and complexity) of both functions by doing something like:
```go
    ...
    if cloneState != cephFSCloneComplete {
        return cloneState.toError()
    }
    // continue like 'case cephFSCloneComplete:'
    ...
}

// put at the place where type cephFSCloneState is defined
func (cs cephFSCloneState) toError() error {
    switch cs {
    case cephFSCloneComplete:
        return nil
    case cephFSCloneError
        return ErrInvalidClone
    ...
}
```